usersession/userd/launcher.go

```diff
@@ -153,10 +157,26 @@ func checkOnClassic() *dbus.Error {
 	return nil
 }
 
+// see https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#file-naming
+var validDesktopFileName = regexp.MustCompile(`^[A-Za-z-_][A-Za-z0-9-_]*(\.[A-Za-z-_][A-Za-z0-9-_]*)*\.desktop$`)
+
+func schemeHasHandler(scheme string) (bool, error) {
+	cmd := exec.Command("xdg-mime", "query", "default", "x-scheme-handler/"+scheme)
+	out, err := cmd.CombinedOutput()
```

Comment:
Nitpick: it would be more robust to use just `Output()` here, in case the `xdg-mime` program emits some warning on the stderr (or even just some debugging message). For example if the combined output was this:
```
Warning: myapp.desktop: invalid desktop file [<- stderr]
gimp.desktop [<- stdout]
```
the current implementation of this function would return `false`, whereas ideally it should return `true`.

Reply:
I've added a TODO here to consider using Output. TBH despite my attempts I was not able to make xdg-mime log anything to stderr even whe ~/.config/mimeapps.list was clearly malformed or missing.